### PR TITLE
Namespace existing API endpoints to api/v0/*

### PR DESCRIPTION
**Pull request by andrewpbrett**
#### What? Why?

In #6881 @Matt-Yorkley [proposed](https://github.com/openfoodfoundation/openfoodnetwork/issues/6881#issuecomment-779420902) this as a way to keep existing endpoints working and allowing us to develop new ones without worrying about breaking the existing ones.

#### What should we test?

We'll want to do a general test of the app, and also potentially test whether things like Zapier integrations still work with the redirect. 

#### Release notes

Prep for API work

Changelog Category: Technical changes

### CODE REVIEW

**Review by sauloperez — APPROVED** [review]
I'd name it `internal` rather than `legacy` so we don't convey the meaning of being old and deprecated (some endpoints might be just fine) while making it clear this is not meant for 3rd-party consumers, but it's just OCD. What we truly need is the namespace.

File: engines/web/app/controllers/web/api/v0/cookies_consent_controller.rb [hunk 1]
```diff
@@ -0,0 +1,32 @@
+require_dependency 'web/cookies_consent'
+
+module Web
+  module Api
+    module V0
+      class CookiesConsentController < BaseController
```
Comment (Matt-Yorkley) [inline engines/web/app/controllers/web/api/v0/cookies_consent_controller.rb:6]: Nice example of something that doesn't need to be in the public API...

File: config/routes/api.rb [hunk 2]
```diff
@@ -1,80 +1,84 @@
...
         end
+        resources :taxons do
+          member do
+            get :jstree
+          end
+        end
       end
     end
+
+    match '*path', to: redirect("/api/v0/%{path}"), via: :all, constraints: { path: /(?!.*v0).+/ }
```
Comment (Matt-Yorkley) [inline config/routes/api.rb:82]: This is the only bit that concerns me. Can we add a couple of simple tests for the routing? I imagine we'll want to change this at some point when we start adding add the `v1` routes.

Comment (Matt-Yorkley) [reply config/routes/api.rb:82]: I think we can get rid of the `.*` in the regex, and it'll only match the `v0` part at the start of the path and not the whole thing. I wonder if we need to check how `GET` params are handled when passed in the URL as well?

Comment (andrewpbrett) [reply config/routes/api.rb]: > I wonder if we need to check how GET params are handled when passed in the URL as well?

I checked, and they were being ignored. Fixed now: fc5909a8a8ae4a6f37515702cf48e2c5efd49f75

> I think we can get rid of the .* in the regex, and it'll only match the v0 part at the start of the path and not the whole thing

Yep, agree: 33d15d2c886fb229e36e5aecb342aa81e536b5dc

> I imagine we'll want to change this at some point...

Might as well do it now I think: bd112515acecf53e685a5bc7817995e7a31c6ef5

> Can we add a couple of simple tests for the routing?

Definitely agree. Coming soon.

Comment (andrewpbrett) [reply config/routes/api.rb]: Here's the first two I thought of: 4052e2acc2ca077caeeb429f117ae4e407c10f1c I might think of some more, but in writing those I noticed that we're going to have to think about authentication pretty soon, if we haven't already. But I think that's outside the scope of this PR.

**Review by Matt-Yorkley — APPROVED** [review]

File: config/routes/api.rb (hunk 2 above)
Comment (Matt-Yorkley) [reply config/routes/api.rb]: Great :muscle:

**Comment by Matt-Yorkley** [comment]
101 files changed, and touching on almost every part of the app... this will need some serious testing.

**Comment by sauloperez** [comment]
There are a bunch of long lines we should fix.

**Review by mkllnk — APPROVED** [review]
Yes, this needs careful testing. But the approach seems useful.

**Comment by filipefurtad0** [comment]
Hey @andrewpbrett,
I see there are conflicts on this branch - should they be solved before testing?

**Comment by andrewpbrett** [comment]
Yep! I’ll rebase and address them.

**Comment by andrewpbrett** [comment]
@filipefurtad0 should be good to test now, pending a green-ish build.

**Comment by filipefurtad0** [comment]
Thank you for addressing this @andrewpbrett 

Looks like there are still issues with the build:
`An error occurred while loading ./spec/controllers/api/v0/taxons_controller_spec.rb`

Maybe this needs a second look? Should it go back to In Dev?

**Comment by andrewpbrett** [comment]
@filipefurtad0 sorry about that. The lesson is to never trust me if I say "this should fix the build..." 😁 

I made a change that should fix it up - I just missed a place to put `V0` earlier. I did run a few of the specs locally and they are working. So pending a green build this should be ready to test.

**Comment by Matt-Yorkley** [comment]
Needs a rebase. Should this one be `dev-test` instead?

**Comment by filipefurtad0** [comment]
Sure, thanks @Matt-Yorkley - I was going through the commits and the respective places on the app where they are called, but maybe it's a good idea to do this after a dev-test, and have another look, for example, when we stage this again for release testing.

**Comment by RachL** [comment]
@andrewpbrett from delivery train tonight: we discussed about getting one more test in - in order to test the redirect that happens on Zapier's API endpoints. @Matt-Yorkley can probably be more specific than myself on this 😅

**Comment by Matt-Yorkley** [comment]
Hmmm... yeah I was wondering about the potential impact on Zapier, but I'm not sure we can improve the specs there, it'll be down to how Zapier handles the redirect response on the client side. So the question is: do we stage this and try to set up Zapier on staging and try it out, or do we just pull a Bill O'Reilly and _do it live_?

do-it-live

**Comment by sauloperez** [comment]
I guess doing a quick check on staging won't take too long and will prevent the people using Zappier from having a bad time if things stop working.

**Comment by andrewpbrett** [comment]
I'd also say we should at least try a basic connectivity test on staging server, unless that would take more than, say a couple hours of dev time to achieve? I'm not as familiar with setting up Zapier as I could be so I don't have a great sense of how long it would take.

**Comment by Matt-Yorkley** [comment]
Cool, I read through the docs a bit and also did a quick live-test on UK staging to see if Zapier handles redirects nicely when interacting with APIs, and the answer seems to be **yes**.

**Pull request merged**